service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/transport/amqps/AmqpSendHandler.java

```diff
@@ -292,21 +181,22 @@ public void onLinkInit(Event event)
     @Override
     public void onLinkFlow(Event event)
     {
-        if (!messagesToBeSent.isEmpty())
+        if (messageToBeSent != null)
         {
-            org.apache.qpid.proton.message.Message protonMessage = messagesToBeSent.remove();
```

Comment:
Was the length of `messagesToBeSent` always one, or did we support some sort of queuing mechanism?

Reply:
It was always a queue of at most 1 value, weirdly. Since the sender uses one connection per message sent, there was no reason to have a queue here